### PR TITLE
feat: [CCM-7335]: ECS Recommendations UI

### PR DESCRIPTION
**Pull request by rajarshee-harness**
##### Jira Links:

https://harness.atlassian.net/browse/CCM-7335

##### Screenshots:

Screenshot 2022-04-27 at 4 37 38 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by rajarshee-harness** [comment]
retrigger coverage

**Comment by rajarshee-harness** [comment]
retrigger coverage

**Review by rishik30 — DISMISSED** [review]

**Comment by rajarshee-harness** [comment]
trigger cypress

**Review by jyotiarora-harness — DISMISSED** [review]

File: src/modules/75-ce/components/ECSRecommendationDetails/ECSRecommendationDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+          color: ChartColors.BLUE
+        },
+        {
+          color: ChartColors.GREY
+        }
+      ]
+    })
+  }
+
+  /* istanbul ignore next */
```
Comment (abhinavrastogi-harness) [inline src/modules/75-ce/components/ECSRecommendationDetails/ECSRecommendationDetails.tsx:141]: ignoring entire functions to get code coverage 😅 not a good idea

File: src/modules/75-ce/components/RecommendationDetails/RecommendationTabs.tsx [hunk 2]
```diff
@@ -19,15 +19,15 @@ import css from './RecommendationDetails.module.scss'
...
+  setSelectedRecommendation: (newState: RecommendationType) => void
   setCPUReqVal: React.Dispatch<React.SetStateAction<number>>
   setMemReqVal: React.Dispatch<React.SetStateAction<number>>
-  setMemLimitVal: React.Dispatch<React.SetStateAction<number>>
-  costOptimizedSavings: number
-  performanceOptimizedSavings: number
-  currentSavings: number
-  isPerfOptimizedCustomized: boolean
-  isCostOptimizedCustomized: boolean
+  setMemLimitVal?: React.Dispatch<React.SetStateAction<number>>
```
Comment (abhinavrastogi-harness) [inline src/modules/75-ce/components/RecommendationDetails/RecommendationTabs.tsx:25]: `setState` should not be passed around. it makes the code fragile and allows breaking the rules of hooks

Comment (abhinavrastogi-harness) [reply src/modules/75-ce/components/RecommendationDetails/RecommendationTabs.tsx:25]: not a blocker, but something to keep in mind for the future

File: src/modules/75-ce/components/RecommendationHistogram/ECSRecommendationHistogram.tsx [hunk 3]
```diff
@@ -0,0 +1,536 @@
...
+      borderWidth: 0,
+      groupPadding: 0,
+      shadow: false
+    },
+    legend: {
+      enabled: false
+    }
+  }
+
+  /* istanbul ignore next */
```
Comment (abhinavrastogi-harness) [inline src/modules/75-ce/components/RecommendationHistogram/ECSRecommendationHistogram.tsx:77]: ignoring entire functions is a bad idea

**Comment by abhinavrastogi-harness** [comment]
Happy to see that no snapshots were using for testing! assertions all the way 👍

File: src/modules/75-ce/components/ECSRecommendationDetails/ECSRecommendationDetails.tsx (hunk 1 above)
Comment (rajarshee-harness) [reply src/modules/75-ce/components/ECSRecommendationDetails/ECSRecommendationDetails.tsx:141]: Most of the these functions are related to interaction with the histogram (using highcharts) present on the page, which are not that easy to test, so chose to ignore them.

**Comment by rajarshee-harness** [comment]
trigger cypress

**Comment by rajarshee-harness** [comment]
trigger cypress

**Comment by rajarshee-harness** [comment]
trigger cypress

**Comment by rajarshee-harness** [comment]
trigger cypress

**Comment by rajarshee-harness** [comment]
trigger cypress

**Comment by rajarshee-harness** [comment]
trigger cypress

**Comment by rajarshee-harness** [comment]
trigger cypress

**Review by jyotiarora-harness — APPROVED** [review]

File: src/modules/10-common/RouteDefinitions.ts [hunk 4]
```diff
@@ -1438,6 +1438,10 @@ const routes = {
   toCEPerspectiveDashboard: withAccountId(() => `/ce/perspective`),
   toCEAnomalyDetection: withAccountId(() => `/ce/anomaly-detection`),
   toBusinessMapping: withAccountId(() => `/ce/cost-categories/`),
+  toCEECSRecommendationDetails: withAccountId(
+    ({ recommendation, recommendationName }: { recommendation: string; recommendationName: string }) =>
+      `/ce/ecs-recommendations/${recommendation}/name/${recommendationName}/details`
```
Comment (jenil-harness) [inline src/modules/10-common/RouteDefinitions.ts:1443]: Will have to change url to `/ce/recommedations/ecs` pattern

**Review by jenil-harness — APPROVED** [review]

**Pull request merged**